logicalplan/codec.go

```diff
@@ -54,6 +56,13 @@ func unmarshalNode(data []byte) (Node, error) {
 		if err := json.Unmarshal(t.Data, v); err != nil {
 			return nil, err
 		}
+		var err error
+		for i, m := range v.LabelMatchers {
+			v.LabelMatchers[i], err = labels.NewMatcher(m.Type, m.Name, m.Value)
```

Comment:
We could explicitly check for regex types, but recreating everything might be more future proof.